Index: app/Modules/Batdongsansg24h/Controllers/PropertyController.php
===================================================================
--- app/Modules/Batdongsansg24h/Controllers/PropertyController.php	(revision 54)
+++ app/Modules/Batdongsansg24h/Controllers/PropertyController.php	(revision 55)
@@ -35,7 +35,7 @@
 
         $user = Auth::guard('bds_sg24h.users')->user();
         $allowEdit = (isset($user) && ($user->id == $prop->owner_user_id || $user->phone == $prop->owner_user_phone));
-        $allowEdit = true;
+//        $allowEdit = true;
 
         $relate_props = null;
         $area_info = [];
Index: app/Modules/Batdongsansg24h/Views/partials/partialProperties.blade.php
===================================================================
--- app/Modules/Batdongsansg24h/Views/partials/partialProperties.blade.php	(revision 54)
+++ app/Modules/Batdongsansg24h/Views/partials/partialProperties.blade.php	(revision 55)
@@ -20,7 +20,7 @@
             </div>
             @if(isset($isEdit)&&$isEdit)
             <div class="row">
-                <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[$prop->owner_user_id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-right" style="text-decoration: none; margin-top: 5px"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
+                <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[\Auth::guard('bds_sg24h.users')->user()->id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-right" style="text-decoration: none; margin-top: 5px"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
             </div>
             @endif
         </div>
Index: app/Modules/Batdongsansg24h/Views/property/info.blade.php
===================================================================
--- app/Modules/Batdongsansg24h/Views/property/info.blade.php	(revision 54)
+++ app/Modules/Batdongsansg24h/Views/property/info.blade.php	(revision 55)
@@ -49,7 +49,7 @@
 @section('main-content-section')
     @if(isset($allowEdit)&&$allowEdit)
         <div class="row">
-            <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[$prop->owner_user_id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-left" style="text-decoration: none; margin-bottom: 10px;"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
+            <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[\Auth::guard('bds_sg24h.users')->user()->id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-left" style="text-decoration: none; margin-bottom: 10px;"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
         </div>
     @endif
 
@@ -147,7 +147,7 @@
     {{--End Property specific--}}
     @if(isset($allowEdit)&&$allowEdit)
         <div class="row">
-            <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[$prop->owner_user_id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-left" style="text-decoration: none; margin-top: 10px;"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
+            <div class="col-xs-12"><a href="{{URL::route('Batdongsansg24hUser.edit_postMsg',[\Auth::guard('bds_sg24h.users')->user()->id,$prop->owner_user_phone,$prop->id])}}" class="btn-sm btn-primary pull-left" style="text-decoration: none; margin-top: 10px;"><i class="ace-icon fa fa-edit icon-animated-vertical"></i> Chỉnh sửa bài viết</a></div>
         </div>
     @endif
 
Index: app/Modules/Batdongsansg24h/Views/user/postMsg.blade.php
===================================================================
--- app/Modules/Batdongsansg24h/Views/user/postMsg.blade.php	(revision 54)
+++ app/Modules/Batdongsansg24h/Views/user/postMsg.blade.php	(revision 55)
@@ -152,7 +152,7 @@
                                         Xem lại
                                     </a>
 
-                                    <a href="{{isset($isEdit)&&$isEdit?URL::route('Batdongsansg24hUserProfile.postedMsg',[$prop->owner_user_id]):URL::route('Batdongsansg24hHome.index')}}" class="btn" type="reset">
+                                    <a href="{{isset($isEdit)&&$isEdit?URL::route('Batdongsansg24hUserProfile.postedMsg',[\Auth::guard('bds_sg24h.users')->user()->id]):URL::route('Batdongsansg24hHome.index')}}" class="btn" type="reset">
                                         <i class="ace-icon fa bigger-110"></i>
                                         Hủy
                                     </a>
